src/resolve/resolvectl.c

```diff
@@ -1241,7 +1242,7 @@ static int read_dns_server_one(sd_bus_message *m, ReadDnsServerFlag flag, char *
                 return 1;
         }
 
-        if (FLAGS_SET(flag, READ_DNS_IFINDEX) && ifindex != 0) {
+        if (FLAGS_SET(flag, READ_DNS_IFINDEX | READ_DNS_IGNORE_NONZERO_IFINDEX) && ifindex != 0) {
```

Comment:
hmm, why hide these DNS servers at all? i mean, we we should always show all DNS servers here, no? but we should not show the ifindex in `%` syntax unless the address from the ipv6 link-local range.

Reply:
I think there's confusion between two related concepts: servers attached to a specific interface and "global" servers, and link-local addresses with an interface specified and global addresses. A "global" server with a link-local address doesn't make much sense: it can only be used if that particular interface is up. Anyway, we try to express both concepts using one interface number.

Reply:
@yuwata ?